### PR TITLE
Borsh in view

### PR DESCRIPTION
**Pull request by mfornet**
Add support for custom way to stringify arguments.

The motivation was to support borsh both in view & call functions.

The implementation is based on @MaximusHaximus suggestion at https://github.com/near/near-api-js/pull/620#discussion_r656438592

### CODE REVIEW

File: src/transaction.ts [hunk 1]
```diff
@@ -51,6 +51,12 @@ export function deployContract(code: Uint8Array): Action {
     return new Action({ deployContract: new DeployContract({code}) });
 }
 
+function stringifyJsonOrBytes(args: any): Buffer {
+    const isUint8Array = args.byteLength !== undefined && args.byteLength === args.length;
+    const serializedArgs = isUint8Array ? args : Buffer.from(JSON.stringify(args));
+    return serializedArgs;
+}
+
```
Comment (mfornet) [inline src/transaction.ts:59]: I created this default function for being backward compatible.

File: src/account.ts [hunk 2]
```diff
@@ -414,11 +422,12 @@ export class Account {
         });
     }
 
-    private functionCallV2({ contractId, methodName, args = {}, gas = DEFAULT_FUNCTION_CALL_GAS, attachedDeposit, walletMeta, walletCallbackUrl }: FunctionCallOptions): Promise<FinalExecutionOutcome> {
+    private functionCallV2({ contractId, methodName, args = {}, gas = DEFAULT_FUNCTION_CALL_GAS, attachedDeposit, walletMeta, walletCallbackUrl, stringify }: FunctionCallOptions): Promise<FinalExecutionOutcome> {
         this.validateArgs(args);
+        const opts = stringify === undefined ? {} : {stringify};
```
Comment (volovyk-s) [inline src/account.ts:427]: let's use a more specific name instead of `opts`

**Review by volovyk-s — COMMENTED** [review]
Great work @mfornet ! Can you please add a simple test that will cover this functionality?

File: src/transaction.ts [hunk 3]
```diff
@@ -59,12 +65,10 @@ export function deployContract(code: Uint8Array): Action {
...
+ * @param stringify Convert input arguments into bytes array.
  */
-export function functionCall(methodName: string, args: Uint8Array | object, gas: BN, deposit: BN): Action {
-    const anyArgs = args as any;
-    const isUint8Array = anyArgs.byteLength !== undefined && anyArgs.byteLength === anyArgs.length;
-    const serializedArgs = isUint8Array ? args : Buffer.from(JSON.stringify(args));
-    return new Action({functionCall: new FunctionCall({methodName, args: serializedArgs, gas, deposit }) });
+export function functionCall(methodName: string, args: Uint8Array | object, gas: BN, deposit: BN, {stringify = stringifyJsonOrBytes} = {}): Action {
+    return new Action({functionCall: new FunctionCall({methodName, args: stringify(args), gas, deposit }) });
 }
```
Comment (MaximusHaximus) [inline src/transaction.ts:72]: There's actually a destructured new object argument aded here that has a single property called `stringify` on that object, which makes the JSDoc here inaccurate.  For the JSDoc to be accurate, function signature should be:
```suggestion
export function functionCall(methodName: string, args: Uint8Array | object, gas: BN, deposit: BN, stringify = stringifyJsonOrBytes): Action {
    return new Action({functionCall: new FunctionCall({methodName, args: stringify(args), gas, deposit }) });
}
```
Or we could document `opts` in the JSDoc

**Review by MaximusHaximus — CHANGES_REQUESTED** [review]
Just one thing about the change to `functionCall()` signature

File: src/account.ts (hunk 2 above)
Comment (mfornet) [reply src/account.ts:427]: is `options` specific enough? 

Feel free to suggest a new name.

Comment (mfornet) [reply src/account.ts]: Removed this altogether.

File: src/transaction.ts (hunk 3 above)
Comment (mfornet) [reply src/transaction.ts]: Applied suggestion instead (use argument stringify)

**Review by volovyk-s — APPROVED** [review]

**Review by MaximusHaximus — APPROVED** [review]
:+1: LGTM :), thanks @mfornet !

**Pull request merged**